### PR TITLE
DOC: rework documentation

### LINKED ISSUE #140: DOC: Update documentation to reflect the new lightpath interface
## Expected Behavior
Lightpath documentation was passed over during #133, and is sorely out of date now.  Will need to go over new lightpath interface and probably provide some examples

## Your Environment
pcds-5.4.1 + #133

### PR DESCRIPTION
**Pull request by tangkong**
## Description
Rewrites documentation to reflect recent changes
updates doctr deploy keys for repository

## Motivation and Context
Docs were out of date.  Closes #140 

## How Has This Been Tested?
I build the docs a lot locally.  Hopefully travis can as well

## Where Has This Been Documented?
This IS documentation

### CODE REVIEW

**Comment by tangkong** [comment]
Not sure how people usually review docs, I've built these locally so if you want to wanted to read them you could I suppose?

`/cds/home/r/roberttk/devrepos/lightpath/docs/build/html`

File: docs/source/interface.rst [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+        # if we were to only care about one signal
+        lightpath_summary = Cpt(EpicsSignal, ':MY:PV')
+
+        input_branches = ['K0']
+        output_branches = ['K0']
+
+        def get_lightpath_state(self):
+            return LightpathState(
+                inserted = True,
```
Comment (klauer) [inline docs/source/interface.rst:120]: Formatting (`inserted=True`)

File: docs/source/interface.rst [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                removed=self._removed,
+                transmission=self._transmission,
+                output_branch=self.output_branches[0]
+            )
+
+In this case we are leveraging the ``LightpathMixin`` class, which does most of
+the repetitive setup for us (creating ``lightpath_summary`` signal, subscribing
+to relevant components, setting up lightpath state caching, checking that the
+subclass is correctly configured, etc.).  This mixin delegates the calculation
+of the lightpaht state to the ``calc_lightpath_state`` method, which is to be
```
Comment (klauer) [inline docs/source/interface.rst:175]:
```suggestion
of the lightpath state to the ``calc_lightpath_state`` method, which is to be
```

File: lightpath/controller.py [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 :class:`.LightController` is what does the organization of all of LCLS's
-devices. After parsing through all of the given devices, each beamline is
-contsructed as a :class:`.BeamPath` object. This includes not only devices on
-the upstream beamline but all of the beamlines before it. For example, the MEC
-beampath will include devices in both the FEE and the XRT. The
-:class:`.LightController` handles this logic as well as a basic overview of
-where the beam is and what the state of the MPS system is currently.
+devices. The facility is represented by an Direced Graph, starting at the
+source and ending at the beamline hutches.  After parsing through all of the
+given devices, each beamline is contsructed as a :class:`.BeamPath` object.
```
Comment (klauer) [inline lightpath/controller.py:6]:
```suggestion
given devices, each beamline is constructed as a :class:`.BeamPath` object.
```

Also, `parsing` (above) might not be the most appropriate word if I understand the section correctly

File: lightpath/path.py [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
-representing the path between them as single object. While the manipulation of
-each of these object should be done at the device level, the
-:meth:`.BeamPath.clear` does provide a powerful tool to quickly change the
-status of the path.
+The :class:`.BeamPath` is the main abstraction for the Lightpath module,
+grouping together a set of devices using the :ref:`Lightpath
+interface<interface_api>` and representing the path between them as single
+object. The manipulation of each of these object should be done at the device
+level, and while this may be done inside of Lightpath via detailed screens,
+Lightpath is not repsonsible for inserting or removing devices.
```
Comment (klauer) [inline lightpath/path.py:7]:
```suggestion
Lightpath is not responsible for inserting or removing devices.
```

**Review by klauer — APPROVED** [review]
I like the new docs quite a bit! A few minor typos here and there is all I noticed.

File: lightpath/path.py (hunk 4 above)
Comment (tangkong) [reply lightpath/path.py:7]: 🤦

File: docs/source/interface.rst [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+What does your ophyd device need?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the simplest sense, your device needs to present the following API:
+
+* ``device.lightpath_summary``: a signal that changes whenever the
+  Lightpath-state of the device changes
+* ``device.get_lightpath_state()``: a method that returns a
+  ``lightpath.LightpathState`` dataclass, containing the Lightpath-state
```
[the remark this answers is not in the document]
Comment (tangkong) [reply docs/source/interface.rst]: Good reminder to include that dataclass in the documentation

**Review by ZLLentz — APPROVED** [review]
I read through the diffs here mainly aiming for accuracy + typos and it all looks good to me. Great docs update!

I did not read through the pre-existing docs

**Comment by tangkong** [comment]
Thanks for the green stamps!  Any further typos will reveal themselves in due time I'm sure 😄

**Pull request merged**